keep/rulesengine/rulesengine.py

```diff
@@ -132,40 +132,65 @@ def run_rules(self, events: list[AlertDto]):
 
             group_status = self._calc_group_status(group.alerts)
             # get the payload of the group
-            # todo: this is not scaling, needs to find another solution
-            # group_payload = self._generate_group_payload(group.alerts)
-            # create the alert
-            event = {
+            # # todo: this is not scaling, needs to find another solution
+            # # group_payload = self._generate_group_payload(group.alerts)
+            # # create the alert
+            # event = {
+            #     "name": group_name,
+            #     "id": group_fingerprint,
+            #     "description": group_description,
+            #     "lastReceived": group_attributes.get("last_update_time"),
+            #     "severity": group_severity,
+            #     "source": group_source,
+            #     "status": group_status,
+            #     "pushed": True,
+            #     "group": True,
+            #     # "groupPayload": group_payload,
+            #     "fingerprint": group_fingerprint,
+            #     **group_attributes,
+            # }
+            # group_alert = create_alert_db(
+            #     tenant_id=self.tenant_id,
+            #     provider_type="group",
+            #     provider_id=rule.id,
+            #     # todo: event should support list?
+            #     event=event,
+            #     fingerprint=group_fingerprint,
+            # )
+            # elastic_client = ElasticClient(self.tenant_id)
+            # elastic_client.index_alert(
+            #     alert=AlertDto(event_id=str(group_alert.id), **event),
+            # )
+            # grouped_alerts.append(group_alert)
+            # self.logger.info(f"Created alert {group_alert.id} for group {group.id}")
+            
+            # Create the incident
+            incident_data = {
                 "name": group_name,
-                "id": group_fingerprint,
+                "group_fingerprint": group_fingerprint,
+                "group_status": group_status,
+                "group_severity": group_severity,
+                "last_received": group_attributes.get("last_update_time"),
                 "description": group_description,
-                "lastReceived": group_attributes.get("last_update_time"),
-                "severity": group_severity,
-                "source": group_source,
-                "status": group_status,
-                "pushed": True,
-                "group": True,
-                # "groupPayload": group_payload,
-                "fingerprint": group_fingerprint,
-                **group_attributes,
+                "is_predicted": True,
             }
-            group_alert = create_alert_db(
+            incident = create_incident_from_dict(
```

Comment:
One question I had here, I suppose we don't want to create a new incident for each new trigger of alert group, rather we want to add new alerts to the same incident we created for the initial trigger. Is my understanding correct, i.e. there should be some new logic for checking whether incident with specific fingerprint already exists, or is it something else expected here?